src/opm/parser/eclipse/Parser/ParserItem.cpp

```diff
@@ -50,14 +52,26 @@ template<> const std::string& default_value< std::string >() {
     return value;
 }
 
-type_tag get_type_json( const std::string& str ) {
+type_tag get_data_type_json( const std::string& str ) {
     if( str == "INT" )       return type_tag::integer;
     if( str == "DOUBLE" )    return type_tag::fdouble;
     if( str == "STRING" )    return type_tag::string;
     if( str == "RAW_STRING") return type_tag::string;
+    if( str == "UDA")        return type_tag::fdouble;
     throw std::invalid_argument( str + " cannot be converted to enum 'tag'" );
 }
 
+/*
+  For very small numbers std::to_string() will just return the string "0.000000"
+*/
+std::string as_string(double value) {
+    if (std::fabs(value) < 1e-4) {
+        std::ostringstream ss;
+        ss << std::setprecision(12) << std::fixed << value;
```

Comment:
After some hunting I discovered a bug here: A value of `1e-20` (which now enters as a default value for a field) will return a `0.00000000000` string rather than `1e-20` due to `std::fixed`. Removing it resolves the issue. I will add a fix in the PR which triggered it.

If you have a reason for using `std::fixed` that I don't see we must find a different solution. I do not see the need at all for this function though, where it is used I think simply `... << setprecision(12) << value ...` would do (although if you need to lower the precision again for some reason you would have to do that).